packages/manager/core/generator/app/templates/src/pages/layout.tsx.hbs

```diff
@@ -1,31 +1,45 @@
 import React, { useEffect, useContext } from 'react';
 import { defineCurrentPage } from '@ovh-ux/request-tagger';
 import { Outlet, useLocation, useMatches } from 'react-router-dom';
-import { useRouting } from '@ovh-ux/manager-react-shell-client';
 import { ShellContext } from '@ovh-ux/manager-react-shell-client';
+import {
+  useOvhTracking,
+  useRouteSynchro,
+} from '@ovh-ux/manager-react-shell-client';
+import { LEVEL2 } from './../tracking.constant';
 
 export default function Layout() {
   const location = useLocation();
-  const routing = useRouting();
   const { shell } = useContext(ShellContext);
   const matches = useMatches();
+  const { environment } = shell;
+  const { trackCurrentPage } = useOvhTracking();
+  useRouteSynchro();
 
   useEffect(() => {
     const match = matches.slice(-1);
     defineCurrentPage(`app.{{appName}}-${match[0]?.id}`);
   }, [location]);
 
   useEffect(() => {
-    shell.ux.hidePreloader();
+    const setTrackingRegion = async () => {
+      try {
+        const env = await environment.getEnvironment();
+        shell.tracking.setConfig(env.getRegion(), LEVEL2);
+      } catch (error) {
+        throw error('layout.tsx setTrackingRegion :', error);
+      }
+    };
+    setTrackingRegion();
   }, []);
 
   useEffect(() => {
-    routing.stopListenForHashChange();
-  }, []);
+    trackCurrentPage();
+  }, [location]);
 
   useEffect(() => {
-    routing.onHashChange();
-  }, [location]);
+    shell.ux.hidePreloader();
+  }, []);
```

Comment:
Why not use the useEffect that already exist ?

Reply:
To separate the logic